pkg/diff/chunk.go

```diff
@@ -221,170 +205,57 @@ func (s *randomSpliter) split(table *TableInstance, columns []*model.ColumnInfo,
 	s.collation = collation
 
 	// get the chunk count by data count and chunk size
-	cnt, err := dbutil.GetRowCount(context.Background(), table.Conn, table.Schema, table.Table, limits)
+	cnt, err := dbutil.GetRowCount(context.Background(), table.Conn, table.Schema, table.Table, limits, nil)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
 	chunkCnt := (int(cnt) + chunkSize - 1) / chunkSize
-	chunks, err := s.splitRange(table.Conn, NewChunkRange(normalMode), chunkCnt, table.Schema, table.Table, columns)
+	chunks, err := splitRangeByRandom(table.Conn, NewChunkRange(), chunkCnt, table.Schema, table.Table, columns, s.limits, s.collation)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
 	return chunks, nil
 }
 
-// splitRange splits a chunk to multiple chunks.
-func (s *randomSpliter) splitRange(db *sql.DB, chunk *ChunkRange, count int, schema string, table string, columns []*model.ColumnInfo) ([]*ChunkRange, error) {
-	var chunks []*ChunkRange
-
+// splitRangeByRandom splits a chunk to multiple chunks by random
+func splitRangeByRandom(db *sql.DB, chunk *ChunkRange, count int, schema string, table string, columns []*model.ColumnInfo, limits, collation string) (chunks []*ChunkRange, err error) {
```

Comment:
I hope you can test the efficiency of splitting under various cases.

Reply:
because I don't know whether it's better

Reply:
ok, I will do a performance test

Reply:
I do a test, table with one hundred million rows, had 204 buckets in tidb's statistical information.
set the chunk-size to 1000,  diff get random value 209 times, and total cost 1m50s to split chunk.